File: fixtures/lifecycle-policy-change/sdk-probe.py
#!/usr/bin/env python3
"""Records which APIs the PyPI reference SDK actually exposes for this family.

Prints one line per probe: supported, or not_supported with the reason. No
assertions about behaviour, only reachability.

    python3 fixtures/lifecycle-policy-change/sdk-probe.py
"""
from __future__ import annotations

import importlib
import importlib.metadata as metadata

PROBES = [
    ("policy version registry, create", "agent_passport", "create_policy_version"),
    ("policy version registry, resolve at instant", "agent_passport", "resolve_policy_version_at"),
    ("operative pointer, issue", "agent_passport", "issue_operative_policy_pointer"),
    ("operative pointer, resolve", "agent_passport", "resolve_operative_policy_pointer"),
    ("policy rollback, classify", "agent_passport", "classify_policy_rollback"),
    ("decision record, render against pinned version", "agent_passport", "render_decision_against_policy_version"),
    ("policy bundle, create", "agent_passport", "create_policy_bundle"),
    ("policy bundle, verify", "agent_passport", "verify_policy_bundle"),
    ("policy chain, verify", "agent_passport", "verify_policy_chain"),
    ("policy decision, verify", "agent_passport", "verify_policy_decision"),
    ("scope version hash, compute", "agent_passport", "compute_scope_version_hash"),
    ("authority chain, verify", "agent_passport.v2.authority_delegation", "verify_authority_delegation_chain"),
    ("authority delegation, issue root", "agent_passport", "issue_authority_delegation"),
    ("RFC 8785 JCS canonical bytes", "agent_passport", "canonicalize_jcs"),
    ("Ed25519 verify", "agent_passport.crypto", "verify"),
]


def main() -> None:
    version = metadata.version("agent-passport-system")
    print(f"agent-passport-system=={version} (PyPI, Python reference SDK)")
    not_supported = 0
    for label, module_name, attribute in PROBES:
        try:
            module = importlib.import_module(module_name)
        except ModuleNotFoundError as error:
            not_supported += 1
            print(f"  not_supported  {label:<46} {module_name}: {error}")
            continue
        if callable(getattr(module, attribute, None)):
            print(f"  supported      {label:<46} {module_name}.{attribute}")
        else:
            not_supported += 1
            print(f"  not_supported  {label:<46} {module_name}.{attribute}: not present in the installed package")
    print("")
    print(f"{len(PROBES) - not_supported}/{len(PROBES)} supported. The policy-version, operative-pointer,")
    print("rollback-classification and decision-rendering steps this family decides are implemented")
    print("in verify.py, not by the SDK. See README.")


if __name__ == "__main__":
    main()

File: fixtures/lifecycle-policy-change/sdk-probe.mjs
// Copyright 2026 Tymofii Pidlisnyi. Apache-2.0 license. See LICENSE.
//
// Records which APIs the npm reference SDK actually exposes for this family.
// Prints one line per probe: supported, or not_supported with the reason. No
// assertions about behaviour, only reachability.
//
//     node fixtures/lifecycle-policy-change/sdk-probe.mjs

import fs from 'node:fs'
import path from 'node:path'
import { fileURLToPath } from 'node:url'

// The package does not export ./package.json, so the version is read from the
// installed tree rather than required.
const here = path.dirname(fileURLToPath(import.meta.url))
const version = JSON.parse(
  fs.readFileSync(path.join(here, '..', '..', 'node_modules', 'agent-passport-system', 'package.json'), 'utf8'),
).version
const sdk = await import('agent-passport-system')

const PROBES = [
  ['policy version registry, create', 'createPolicyVersion'],
  ['policy version registry, resolve at instant', 'resolvePolicyVersionAt'],
  ['operative pointer, issue', 'issueOperativePolicyPointer'],
  ['operative pointer, resolve', 'resolveOperativePolicyPointer'],
  ['policy rollback, classify', 'classifyPolicyRollback'],
  ['decision record, render against pinned version', 'renderDecisionAgainstPolicyVersion'],
  ['decision record, read pinned policy version', 'policyVersionOfDecision'],
  ['policy bundle, create', 'createPolicyBundle'],
  ['policy bundle, verify', 'verifyPolicyBundle'],
  ['policy chain, verify', 'verifyPolicyChain'],
  ['policy receipt, create', 'createPolicyReceipt'],
  ['policy decision, verify', 'verifyPolicyDecision'],
  ['scope version hash, compute', 'computeScopeVersionHash'],
  ['scope version match, verify', 'verifyScopeVersionMatch'],
  ['authority chain, verify', 'verifyAuthorityDelegationChain'],
  ['authority delegation, issue root', 'issueAuthorityDelegation'],
  ['RFC 8785 JCS canonical bytes', 'canonicalizeJCS'],
  ['Ed25519 verify', 'verify'],
]

console.log(`agent-passport-system@${version} (npm, TypeScript reference SDK)`)
let notSupported = 0
for (const [label, name] of PROBES) {
  if (typeof sdk[name] === 'function') {
    console.log(`  supported      ${label.padEnd(46)} ${name}`)
  } else {
    notSupported += 1
    console.log(`  not_supported  ${label.padEnd(46)} ${name}: not exported from the package root`)
  }
}
console.log('')
console.log(
  `${PROBES.length - notSupported}/${PROBES.length} supported. The policy-version, operative-pointer, rollback-classification`,
)
console.log(
  'and decision-rendering steps this family decides are implemented in harness.ts, not by the SDK. See README.',
)
